pkg/controller/configmapcainjector/configmap_injecting_controller.go

```diff
@@ -167,8 +167,13 @@ func (ic *ConfigMapCABundleInjectionController) syncConfigMap(key string) error
 		return nil
 	}
 
-	configMapCopy := sharedConfigMap.DeepCopy()
 	// make a copy to avoid mutating cache state
+	configMapCopy := sharedConfigMap.DeepCopy()
+
+	if configMapCopy.Data == nil {
```

Comment:
Do we need to check this for the `sharedConfigMap` a few lines up as well?

Reply:
No, it's only assignment to nil map that breaks us here, the above will return !ok.